### PR TITLE
[Concept Entry] AI Convolutional Neural Networks (CNN)

### PR DESCRIPTION
**Pull request by kapoorsaumitra**
### Description

Created convolutional-neural-networks.md for #2016 

### Type of Change

- Adding a new entry

### Checklist

- [x] All writings are my own.
- [x] My entry follows the Codecademy Docs style guide.
- [x] My changes generate no new warnings.
- [x] I have performed a self-review of my own writing and code.
- [x] I have checked my entry and corrected any misspellings.
- [x] I have made corresponding changes to the documentation if needed.
- [x] I have confirmed my changes are not being pushed from my forked `main` branch.
- [x] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [x] Under "Development" on the right, I have linked any issues that are relevant to this PR (write "Closes #<issue number> in the "Description" above).

### CODE REVIEW

File: content/ai/concepts/neural-networks/terms/convolutional-neural-networks/convolutional-neural-networks.md
Comment (SSwiniarski) [inline content/ai/concepts/neural-networks/terms/convolutional-neural-networks/convolutional-neural-networks.md:12]:
```suggestion
```
CNN is not on the [tags list](https://github.com/Codecademy/docs/blob/main/documentation/tags.md).

File: content/ai/concepts/neural-networks/terms/convolutional-neural-networks/convolutional-neural-networks.md
Comment (SSwiniarski) [inline content/ai/concepts/neural-networks/terms/convolutional-neural-networks/convolutional-neural-networks.md:20]:
```suggestion
**Convolutional Neural Networks** (CNNs) are a type of neural network that are primarily used for image classification. They are also used for other tasks such as object detection, image segmentation, and natural language processing.
```
Spell out an acronym the first time its used. And we bold the first appearance of the entry term.

File: content/ai/concepts/neural-networks/terms/convolutional-neural-networks/convolutional-neural-networks.md
Comment (SSwiniarski) [inline content/ai/concepts/neural-networks/terms/convolutional-neural-networks/convolutional-neural-networks.md:22]:
```suggestion
CNNs are inspired by the visual cortex of the brain. The visual cortex is the part of the brain that processes visual information. 
```
We don't bold new terms in Docs.

File: content/ai/concepts/neural-networks/terms/convolutional-neural-networks/convolutional-neural-networks.md
Comment (SSwiniarski) [inline content/ai/concepts/neural-networks/terms/convolutional-neural-networks/convolutional-neural-networks.md:24]:
```suggestion
## How does a CNN work?
```
We don't use h1 headers. That's reserved for the title provided by the metadata.

File: content/ai/concepts/neural-networks/terms/convolutional-neural-networks/convolutional-neural-networks.md
Comment (SSwiniarski) [inline content/ai/concepts/neural-networks/terms/convolutional-neural-networks/convolutional-neural-networks.md:26]:
```suggestion
CNNs extract features from images using specialized layers called convolutional layers. These layers apply filters to the input image which helps them to learn about features such as patterns and edges.
```

File: content/ai/concepts/neural-networks/terms/convolutional-neural-networks/convolutional-neural-networks.md
Comment (SSwiniarski) [inline content/ai/concepts/neural-networks/terms/convolutional-neural-networks/convolutional-neural-networks.md:18]:
```suggestion
```
Remove the first header, we don't have headers before the introductory paragraph.

File: content/java/concepts/treemap/treemap.md
Comment (SSwiniarski) [inline content/java/concepts/treemap/treemap.md:1]: It seems you either created this branch off of another branch, or tried to make two PRs off of the same branch. Can we undo these edits so this PR is solely about CNNs?

File: content/ai/concepts/neural-networks/terms/convolutional-neural-networks/convolutional-neural-networks.md
Comment (SSwiniarski) [inline content/ai/concepts/neural-networks/terms/convolutional-neural-networks/convolutional-neural-networks.md:32]:
```suggestion
Due to their ability to extract features from images, CNNs are widely used in the field of computer vision with its applications ranging from augmented-reality to self-driving cars.
```

**Review by SSwiniarski — CHANGES_REQUESTED** [review]
@kapoorsaumitra I did a first review and left a number of comments. I also note that you seem to have another PR mixed in with this one. The edits to the Java TreeMap entry belong in their own PR.

File: content/java/concepts/treemap/treemap.md
Comment (kapoorsaumitra) [reply content/java/concepts/treemap/treemap.md:1]: i'll create a new PR with the suggested changes.

Comment (kapoorsaumitra) [reply content/java/concepts/treemap/treemap.md:1]: i've updated the PR, please have a look.

**Review by SSwiniarski — APPROVED** [review]
@kapoorsaumitra I did a couple of minor edits and this now looks like it's ready for a second review.

**Comment by SSwiniarski** [comment]
@kapoorsaumitra FYI, the second review has to come from another maintainer. I already approved this. I went ahead and flagged the other maintainers so one of them should come check on this.

**Review by KTom101 — APPROVED** [review]
@kapoorsaumitra Great entry. I've added some minor changes directly which include:

- Changing "Deep Learning" to "Machine Learning" in the metadata because "Deep Learning" wasn't on the [Subjects list](https://github.com/Codecademy/docs/blob/main/documentation/subjects.md) and the concept entry uses "Machine Learning" already so I stuck with that.
- Converted "How does a CNN work?" subheading into AP title-case
- Running `npm run format:all` to get the checks to pass

**Comment by KTom101** [comment]
@kapoorsaumitra Entry is live: https://www.codecademy.com/resources/docs/ai/neural-networks/convolutional-neural-networks

Screenshot 2023-06-16 at 4 40 43 PM

cc @yangc95

**Pull request merged**
